webapp/components/wpt-processor.js

```diff
@@ -74,26 +93,35 @@ class WPTProcessor extends LoadingState(PolymerElement) {
       testRuns: {
         type: Array
       },
+      resultsLoadFailed: {
+        type: Boolean,
+        value: false,
+      },
+      selectedTab: {
+        type: Number,
+        value: 0,
+        observer: '_selectedTabChanged',
+      }
     };
   }
 
-  constructor() {
-    super();
-    this.onLoadingComplete = () => {
-      this.loadingFailed = !this.testRuns;
-    };
-  }
-
-  async ready() {
-    await super.ready();
+  _selectedTabChanged(tab) {
+    const path = tab === 0 ? '/api/status/pending' : '/api/status/invalid';
     this.load(
-      this.loadPendingRuns()
-    );
+        this.loadPendingRuns(path),
+        () => {
+          this.resultsLoadFailed = true;
+          this.testRuns = [];
+        });
   }
 
-  async loadPendingRuns() {
-    const r = await fetch('/api/status/pending');
-    this.testRuns = r.ok && await r.json();
+  async loadPendingRuns(path) {
+    this.resultsLoadFailed = false;
+    const r = await fetch(path);
+    if (!r.ok || r.status !== 200) {
```

Comment:
https://fetch.spec.whatwg.org/#dom-response-ok checks for status in the range 200 to 299, why do we demand that the status is 200? Also, don't we have a wrapper for this very common pattern of fetch-and-check-status?

Reply:
Changed to checking `r.ok` only. And a quick search didn't find any wrapper. Perhaps we can refactor this into `loading-state.js`. Another day.